search/api.py

```diff
@@ -215,18 +287,39 @@ def execute_search(*, user, query):
     search = Search(index=index, using=get_conn())
     search.update_from_dict(query)
     search = _apply_general_query_filters(search, user)
-    return transform_aggregates(search.execute().to_dict())
+    if not user.is_anonymous and is_learning_query(query):
```

Comment:
This `is_learning_query` is ok for now, but I think this kind of introspection of the incoming query in order to specialize the query further is a sign we need to look into providing an alias for the current search endpoint that applies different query logic for different logical (learning resources vs. posts/comments) searches. It's also impossible to detect when this logic breaks outside of a full integration test between the frontend and backend.

That said, I think this approach is a reasonable tradeoff vs. spending time on a separate endpoint for the time being since things shouldn't change that much here, but we should keep an eye on it and any further changes of a similar nature should probably be the trigger point for such a refactor.